src/Tizen.Wearable.CircularUI.Chart.Forms.Renderer/BarChartViewRenderer.cs

```diff
@@ -0,0 +1,752 @@
+                {
+                    Element.DrawChartRequested -= OnDrawChartRequested;
+                }
+
+                if (Control != null)
+                {
+                    Control.PaintSurface -= OnPaintSurface;
+                }
+            }
+
+            base.Dispose(disposing);
+        }
+
+
+        private void OnPaintSurface(object sender, SKPaintSurfaceEventArgs e)
+        {
+            _canvas = e.Surface.Canvas;
+            _canvasSize.Width = e.Info.Width;
+            _canvasSize.Height = e.Info.Height;
+            _canvas.Clear();
+            DrawContent(_canvas);
+        }
+
+        protected virtual void DrawContent(SKCanvas canvas)
+        {
+            if (Element == null || Element.Data == null)
+            {
+                Log.Debug(FormsCircularUIChart.Tag, "Bar Chart Element == null || Element.Data == null");
+                return;
+            }
+
+            if (Element.AxisOption != null)
+            {
+                CaculateAxisSize(canvas);
+            }
+            else
+            {
+                _majorAxisSize = new SKSize(0, 0);
+                _minorAxisSize = new SKSize(0, 0);
+                _referenceItemSize = new SKSize(0, 0);
+            }
+
+            var barSize = CalculateBarSize();
+            GenerateDataTable();
+            var points = CalculatePoints(barSize);
+            if (Element.AxisOption.IsVisibleOfMajorAxisLine)
+            {
+                DrawMajorAxisLine(canvas);
+            }
+
+            if (Element.AxisOption.IsVisibleOfCategoryLabel)
+            {
+                DrawCategoryLabels(canvas, points);
+            }
+
+            if (Element.AxisOption.IsVisibleOfMinorAxisLine)
+            {
+                DrawMinorAxisLine(canvas);
+            }
+
+            if (Element.AxisOption.IsVisibleOfReferenceLabel)
+            {
+                DrawReferenceLabels(canvas, barSize);
+            }
+
+            if (Element.BarBackgroundColorIsVisible)
+            {
+                DrawBarBackground(canvas, points, barSize);
+            }
+
+            if (Element.AxisOption.IsVisibleOfReferenceLine)
+            {
+                DrawReferenceLines(canvas, barSize);
+            }
+
+            DrawBars(canvas, points, barSize);
+
+            if (Element.ValueLabelIsVisible)
+            {
+                DrawValueLabels(canvas, points, barSize);
+            }
+        }
+
+        protected virtual void GenerateDataTable()
+        {
+            if (Element.Data == _prevDataSetArray)
+                return;
+
+            _barChartDataTable = new DataItem[_categoryCount];
+            var items = Element.Data[0].DataItems;
+            for (int i = 0; i < items.Count; i++)
+            {
+                var index = items[i].Key == 0 ? i : items[i].Key - 1;
+                index = index < _categoryCount ? index : _categoryCount - 1;
+                _barChartDataTable[index] = items[i] as DataItem;
```

Comment:
Why do you cast as DataItem instead of IDataItem? It is not working if User implement a owned class that is implementing IDataItem
```suggestion
                _barChartDataTable[index] = items[i] as IDataItem;
```

Reply:
Done